playbooks/roles/genesis-exoscale/tasks/main.yml

```diff
@@ -0,0 +1,178 @@
+---
+- name: Set the Exoscale API details facts to the value that was entered, or attempt to retrieve it from the environment if the entry is blank
+  set_fact:
+    EXO_API_SECRET: "{{ EXO_API_SECRET_entry | default( lookup('env', 'EXO_API_SECRET') ) }}"
+    EXO_API_KEY: "{{ EXO_API_KEY_entry | default( lookup('env', 'EXO_API_KEY') ) }}"
+    EXO_API_ENDPOINT: "{{ lookup('env', 'EXO_API_ENDPOINT') | default('https://api.exoscale.ch/compute', true) }}"
+
+
+- name: Get the default SSH key
+  command: cat ~/.ssh/id_rsa.pub
+  register: ssh_key
+
+- block:
+    - name: Add the SSH key to Exoscale if it doesn't already exist
+      local_action:
+        module: cs_sshkeypair
+        name: "{{ exo_ssh_name }}"
+        public_key: "{{ ssh_key.stdout }}"
+      register: exo_ssh_key
+  rescue:
+    - fail:
+        msg: "* The API key or secret might be incorrect or missing the Write Scope. OR * The SSH key may already exist in the Exoscale portal under a different name. OR * The cs Python module might not be installed. Use `pip install cs` to install this module."
+
+- block:
+  - name: Streisand Security Group
+    local_action:
+      module: cs_securitygroup
+      name: "streisand"
+      description: streisand VPN
+  
+  - name: public to streisand ssh
```

Comment:
I can clean these up in a separate commit, if you'd like, but I have tried to make the task descriptions very clear and readable. My suggestion for this one, for example, would be:

```yaml
  - name: Open the SSH port in the Exoscale Security Group
```